oslogin: compiling error

I am trying to compile oslogin in a centos7 machine, and I get the error below.
I can fix the issue by reverting commit "Add user name validation to pam modules" #633
I'll take a deeper look, just adding here to track it.

```
[root@koike-centos7 google_compute_engine_oslogin]# make
g++ -fPIC -I/usr/include/json-c -c utils/oslogin_utils.cc -o utils/oslogin_utils.o
In file included from /usr/include/c++/4.8.2/regex:35:0,
                 from utils/oslogin_utils.cc:25:
/usr/include/c++/4.8.2/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support is currently experimental, and must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support for the \
  ^
utils/oslogin_utils.cc: In function ‘bool oslogin_utils::ValidateUserName(const string&)’:
utils/oslogin_utils.cc:230:3: error: ‘regex’ is not a member of ‘std’
   std::regex r(kUserNameRegex);
   ^
utils/oslogin_utils.cc:230:14: error: expected ‘;’ before ‘r’
   std::regex r(kUserNameRegex);
              ^
utils/oslogin_utils.cc:231:10: error: ‘regex_match’ is not a member of ‘std’
   return std::regex_match(user_name, r);
          ^
utils/oslogin_utils.cc:231:38: error: ‘r’ was not declared in this scope
   return std::regex_match(user_name, r);
                                      ^
make: *** [utils/oslogin_utils.o] Error 1
```

--- a/google_compute_engine_oslogin/packaging/rpmbuild/SPECS/google-compute-engine-oslogin.spec
+++ b/google_compute_engine_oslogin/packaging/rpmbuild/SPECS/google-compute-engine-oslogin.spec
@@ -25,12 +25,14 @@ Summary:        OS Login Functionality for Google Compute Engine
 License:        ASL 2.0
 Source0:        %{name}_%{version}.orig.tar.gz
 
+BuildRequires:  boost-devel
 BuildRequires:  gcc-c++
 BuildRequires:  make
 BuildRequires:  libcurl
 BuildRequires:  json-c
 BuildRequires:  pam-devel
 BuildRequires:  policycoreutils-python
+Requires:  boost-regex
 Requires:  policycoreutils-python
 
 %define pam_install_path /%{_lib}/security
@@ -43,7 +45,7 @@ for Google Compute Engine.
 %setup
 
 %build
-make %{?_smp_mflags} LIBS="-lcurl -ljson-c"
+make %{?_smp_mflags} LIBS="-lcurl -ljson-c -lboost_regex"
 
 %install
 rm -rf %{buildroot}

--- a/google_compute_engine_oslogin/utils/oslogin_utils.cc
+++ b/google_compute_engine_oslogin/utils/oslogin_utils.cc
@@ -22,7 +22,19 @@
 #include <cstring>
 #include <iostream>
 #include <sstream>
+
+#ifdef __GNUC__
+#if __GNUC__ > 4 || \
+  (__GNUC__ == 4 && (__GNUC_MINOR__ > 9 || \
+                     (__GNUC_MINOR__ == 9 && \
+                      __GNUC_PATCHLEVEL__ > 0)))
 #include <regex>
+#define Regex std
+#else
+#include <boost/regex.hpp>
+#define Regex boost
+#endif
+#endif
 
 #include "oslogin_utils.h"
 
@@ -227,8 +239,8 @@ bool HttpGet(const string& url, string* response, long* http_code) {
 }
 
 bool ValidateUserName(const string& user_name) {
-  std::regex r(kUserNameRegex);
-  return std::regex_match(user_name, r);
+  Regex::regex r(kUserNameRegex);
+  return Regex::regex_match(user_name, r);
 }
 
 string UrlEncode(const string& param) {

